### PR TITLE
Add public IP to config when btfs init and in new announce-public pro…

### PR DESCRIPTION
**Pull request by jin-tron**
Public IP was added to config Addresses Announce when run btfs daemon.
In this case we cannnot run two daemon on same server because port conflict.
Now we add public IP when btfs init, also provide a new config profile to add
public IP.
  btfs config profile apply announce-public
When run "btfs config profile apply default-networking", the Announce public IP
will be deleted in config.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -116,6 +118,8 @@ require (
 	github.com/prometheus/procfs v0.0.0-20190519111021-9935e8e0588d // indirect
 	github.com/shirou/gopsutil v2.18.12+incompatible
 	github.com/syndtr/goleveldb v1.0.0
+	github.com/tyler-smith/go-bip32 v0.0.0-20170922074101-2c9cfd177564
+	github.com/tyler-smith/go-bip39 v1.0.0
```
Comment (taiyangc) [inline go.mod]: Where do you use these modules?

File: cmd/btfs/init.go [hunk 2]
```diff
@@ -270,12 +270,17 @@ func addDefaultAssets(out io.Writer, repoRoot string) error {
...
+	buildCfg := &core.BuildCfg{Repo: r}
+	nd, err := core.NewNode(ctx, buildCfg)
 	if err != nil {
 		return err
 	}
 	defer nd.Close()
 
+	// announce public ip by default for btfs nodes running on cloud vm
+	// or local network with NAT
+	buildCfg.AnnouncePublicIp()
```
Comment (taiyangc) [inline cmd/btfs/init.go]: Check for err

File: go.mod (hunk 1 above)
Comment (jin-tron) [reply go.mod]: Actually these come from another ticket for seed phrase, might missing from there. @elluck91

File: cmd/btfs/init.go (hunk 2 above)
Comment (jin-tron) [reply cmd/btfs/init.go]: done

**Pull request merged**